packages/sdk-metrics/src/export/MetricReader.ts

```diff
@@ -45,6 +45,11 @@ export interface MetricReaderOptions {
    * not configured, cumulative is used for all instruments.
    */
   aggregationTemporalitySelector?: AggregationTemporalitySelector;
+  /**
+   * Additional MetricProducers to use as a source of aggregated metric data in addition to the
+   * SDK's metric data.
+   */
+  metricProducers?: MetricProducer[];
```

Comment:
Note the specification asks for a `RegisterMetricProducer` function https://github.com/open-telemetry/opentelemetry-specification/blob/v1.23.0/specification/metrics/sdk.md#registerproducermetricproducer

However, I since we already have a public `setMetricProducer` which assumes it is receiving the SDK's `MetricCollector`, this seems like the cleanest solution without breaking anything.

I will also check with the spec on this.